modules/core/client/directives/pagination.client.directive.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+//Pagination Directive
+angular.module('core').directive('meanPagination', function () {
+    return {
+        restrict: 'AEC',
+        scope: {
+            meanPaginationData: '=meanPaginationData',
+            meanPaginationItemCount: '=meanPaginationItemCount',
+            meanPaginationUrl: '@meanPaginationUrl',
+            meanPaginationSearch: '@meanPaginationSearch'
+        },
+        controller: ['$scope', '$filter', function($scope, $filter) {
+
+            $scope.buildPager = function () {
+                $scope.pagedItems = [];
+                $scope.itemsPerPage = $scope.meanPaginationItemCount;
+                $scope.currentPage = 1;
+                $scope.figureOutItemsToDisplay();
+            };
+
+            $scope.figureOutItemsToDisplay = function () {
+                $scope.filteredItems = $filter('filter')($scope.meanPaginationData, {
+                    $: $scope.search
+                });
+                $scope.filterLength = $scope.filteredItems.length;
+                var begin = (($scope.currentPage - 1) * $scope.itemsPerPage);
+                var end = begin + $scope.itemsPerPage;
+                $scope.pagedItems = $scope.filteredItems.slice(begin, end);
+            };
+
+            $scope.pageChanged = function () {
+                $scope.figureOutItemsToDisplay();
+            };
+
+            $scope.$watch('meanPaginationData', function (newValue, oldValue) {
+                if (newValue !== oldValue) {
```

Comment:
Is this for when you have an object/array that is the same content but from a different source?